pyccl/pk2d.py

```diff
@@ -170,6 +170,20 @@ def apply_halofit(Pk2D, cosmo, pk_linear):
             pk_linear (:class:`Pk2D`): a :class:`Pk2D` object containing
                 the linear power spectrum to transform.
         """
+        if cosmo["wa"] != 0:
+            # HALOFIT translates (w0, wa) to a w0_eff. This requires computing
+            # the comoving distance to the CMB, which requires the background
+            # splines being sampled to sufficiently high redshifts.
+            _, a = _get_spline1d_arrays(cosmo.cosmo.data.achi)
```

Comment:
I think there's no guarantee that cosmo will have computed distances yet. Can you add a `cosmo.compute_distances()` above this?

Reply:
Right! Added.